lib/agent0/agent0/hyperdrive/policies/random_agent.py

```diff
@@ -59,14 +57,14 @@ def get_available_actions(
             all_available_actions.append(HyperdriveActionType.CLOSE_SHORT)
         if wallet.lp_tokens:
             all_available_actions.append(HyperdriveActionType.REMOVE_LIQUIDITY)
-        if wallet.withdraw_shares and market.market_state.withdraw_shares_ready_to_withdraw > 0:
+        if wallet.withdraw_shares and interface.pool_info["withdrawSharesReadyToWithdraw"] > 0:
```

Comment:
A bit confused on the logic here, you're checking the global pool_info parameter to see if there are withdraw_shares ready to withdraw for this wallet?

Reply:
it is checking if this wallet has withdraw shares, and if there are withdraw shares that are ready to withdraw in the pool itself. It's possible that the wallet expects to get withdraw shares, but they're still tied up in the pool